app/controllers/users/notification_settings_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Users
+  class NotificationSettingsController < ApplicationController
+    before_action :raise_suspended
+    before_action :authenticate_user!
+    after_action :verify_authorized
+
+    ALLOWED_PARAMS = %i[id
+                        email_badge_notifications
+                        email_comment_notifications
+                        email_community_mod_newsletter
+                        email_connect_messages
+                        email_digest_periodic
+                        email_follower_notifications
+                        email_membership_newsletter
+                        email_mention_notifications
+                        email_newsletter
+                        email_tag_mod_newsletter
+                        email_unread_notifications
+                        mobile_comment_notifications
+                        mod_roundrobin_notifications
+                        reaction_notifications
+                        welcome_notifications
+                        user_id].freeze
```

Comment:
I think if `user_id` is allowed here, someone could theoretically reassign their notification settings to someone else, resulting in that other user having 2 (making their notification settings non-deterministic) and the current user not having one at all, possibly requiring engineer intervention to re-create it for them (assuming we don't recreate them on the fly).

```suggestion
                        welcome_notifications].freeze
```

I'm making an assumption here about not creating the notification setting record, but I tend to think about worst-case scenarios since that's oddly helpful as an SRE — and if I happen to be correct, I'm probably the one that has to intervene. :joy: